nearcore/src/config.rs

```diff
@@ -394,7 +394,7 @@ impl Default for Config {
 }
 
 impl Config {
-    pub fn from_file(path: &Path) -> anyhow::Result<Self> {
+    pub fn from_file(path: &Path, config_validation: ConfigValidationMode) -> anyhow::Result<Self> {
```

Comment:
`config.json` and `node_key.json` should always be validated.
`GenesisValidationMode` is an optimization to improve the node startup time because reading the testnet genesis takes a couple of minutes.
Validating `config.json` and `node_key.json` takes almost zero time.

I suggest keeping `GenesisValidationMode` and not replacing it with `ConfigValidationMode`.